src/data/zha.ts

```diff
@@ -0,0 +1,105 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import { HomeAssistant } from "../types";
+
+export interface ZHADeviceEntity extends HassEntity {
+  device_info?: {
+    identifiers: any[];
+  };
+}
+
+export interface ZHAEntities {
+  [key: string]: HassEntity[];
+}
+
+export interface Attribute {
+  name: string;
+  id: number;
+}
+
+export interface Cluster {
+  name: string;
+  id: number;
+  type: string;
+}
+
+export interface Command {
+  name: string;
+  id: number;
+  type: string;
+}
+
+export interface ReadAttributeServiceData {
+  entity_id: string;
+  cluster_id: number;
+  cluster_type: string;
+  attribute: number;
+  manufacturer: number;
+}
+
+export const reconfigureNode = (
+  hass: HomeAssistant,
+  ieeeAddress: string
+): Promise<void> =>
+  hass.callWS({
+    type: "zha/nodes/reconfigure",
+    ieee: ieeeAddress,
+  });
+
+export const fetchAttributesForCluster = (
+  hass: HomeAssistant,
+  entityId: string,
+  ieeeAddress: string,
+  clusterId: number,
+  clusterType: string
+): Promise<Attribute[]> =>
+  hass.callWS({
+    type: "zha/entities/clusters/attributes",
+    entity_id: entityId,
+    ieee: ieeeAddress,
+    cluster_id: clusterId,
+    cluster_type: clusterType,
+  });
+
+export const readAttributeValue = (
+  hass: HomeAssistant,
+  data: ReadAttributeServiceData
+): Promise<string> => {
+  const serviceData = {
+    type: "zha/entities/clusters/attributes/value",
+  };
+  Object.assign(serviceData, data);
```

Comment:
This is not needed. Did you see my example code in my previous comment?

```ts
hass.callWS({
  ...data,
  type: "zha/entities/clusters/attributes/value",
})
```

Reply:
Whoops I’ll fix it